src/pages/notification/index.vue

```diff
@@ -0,0 +1,343 @@
+        let fromName;
+        let toName;
+        let price;
+        const eventHasSeen = this.checkHasSeenEvent(e.timestamp);
+
+        let memo = e.memo === 'like.co NFT API' ? '' : e.memo;
+        const nftName = this.getNFTClassMetadataById(e.class_id)?.name;
+        if (e.sender === LIKECOIN_NFT_API_WALLET) {
+          if (e.receiver === this.getAddress) {
+            eventType = 'purchase_nft';
+            const creator = this.getNFTClassMetadataById(e.class_id)
+              ?.iscn_owner;
+            displayAvatar = this.getUserInfoByAddress(this.getAddress)?.avatar;
+            i18nPath = 'event_list_page_event_message_purchase_nft';
+            fromName = this.$t('event_list_page_event_self');
+            toName = this.getUserInfoByAddress(creator)?.displayName || creator;
+            // eslint-disable-next-line prefer-destructuring
+            price = e.price;
+            memo = e.granterMemo || '';
+          } else {
+            eventType = 'nft_sale';
+            fromName =
+              this.getUserInfoByAddress(e.receiver)?.displayName || e.receiver;
+            // eslint-disable-next-line prefer-destructuring
+            price = e.price;
+            i18nPath = 'event_list_page_event_message_nft_sale';
+            displayAvatar = this.getUserInfoByAddress(e.receiver)?.avatar;
+            memo = e.granterMemo || '';
+          }
+        } else if (e.receiver === this.getAddress) {
+          eventType = 'receive_nft';
+          fromName =
+            this.getUserInfoByAddress(e.sender)?.displayName || e.sender;
+          i18nPath = 'event_list_page_event_message_receive_nft';
+          displayAvatar = this.getUserInfoByAddress(e.sender)?.avatar;
+        } else {
+          eventType = 'send_nft';
+          fromName = this.$t('event_list_page_event_self');
+          toName =
+            this.getUserInfoByAddress(e.receiver)?.displayName || e.receiver;
+          i18nPath = 'event_list_page_event_message_send_nft';
+          displayAvatar = this.getUserInfoByAddress(e.receiver)?.avatar;
+        }
+        if (memo) {
+          message = memo;
+        }
+
+        return {
+          eventType,
+          displayAvatar,
+          eventHasSeen,
+          message,
+          fromName,
+          toName,
+          price,
+          nftName,
+          i18nPath,
+          targetRoute: {
+            name: e.nft_id ? 'nft-class-classId-nftId' : 'nft-class-classId',
+            params: {
+              classId: e.class_id,
+              nftId: e.nft_id,
+            },
+          },
+          ...e,
+        };
+      });
+    },
+    groupedEventsByTime() {
+      return this.convertToGroupedEvents(this.processedEvents);
+    },
+  },
+  async mounted() {
+    if (this.getAddress && !this.walletHasLoggedIn) {
+      try {
+        await this.signLogin();
+      } catch {
+        // No-op
+      }
+    }
+  },
+  beforeRouteLeave(to, form, next) {
+    this.updateEventLastSeenTs();
+    next();
```

Comment:
Does this trigger on page/browser close?

Reply:
well ~ It doesn‘t 🙃, I changed it to use `beforeDestroy` to make sure it will be triggered when leaving/closing the page in any situation.